### ISSUE TITLE
`Non-local exit from iterator` has a false positive with #define_singleton_method

### ISSUE THREAD

**Issue by AlexWayfer** [bbatsov/rubocop#3746]
Hello!

I have a code:

```ruby
str = 'foo'
str.define_singleton_method :bar do |baz|
  return unless baz
  replace baz
end
```

I think I can make PR for fixing this.

--------

## Expected behavior

No warnings from Rubocop.

## Actual behavior

Warning about `Non-local exit from iterator`.

## Steps to reproduce the problem

Check the file with this code (or similar) by Rubocop.

## RuboCop version

```
$ rubocop -V
0.45.0 (using Parser 2.3.1.4, running on ruby 2.3.3 x86_64-linux)
```
